src/pages/Home.js

```diff
@@ -31,47 +40,112 @@ const Home = () => {
         setAddedProject(false)
     }
 
-    const postNewProject = () => {
+    const postProject = () => {
         Fire
         .firestore()
         .collection("projects")
         .add({
             projectName: projectName,
             userId: user.uid,
         })
-        .then(() => console.log('succes'))
+        .then()
         .catch(error => console.error("Error adding document: ", error))
         setAddedProject(true)
         dispatch(fetchProjects(user.uid));
     }
 
+    const deleteProject = (projectId) => {
+        Fire.firestore().collection('projects').doc(projectId).delete().then(() => {
+            dispatch(fetchProjects(user.uid));
+        })
+        .catch((error) => {
+            console.error("Error removing document: ", error);
+        })
+
+        //also have to delete all tasks related to this project
+    }
+
+    const selectProject = (selected) => {
+        setSelectedProject(selected);
+    }
+
+    //Task-related functions
+    const newTaskHandler = (event) => {
+        setNewTask({
+            ...newTask,
+            [event.target.name]: event.target.value,
+        })
+    }
+
+    const postTask = () => {
+        Fire
```

Comment:
Omda tje firestore een hoop aanspreekt op verschillende punten. kan je overwegen je data access naar de tasks etc in aparte service te stoppen Dao repository pattern.
```
tasks.addTask()
task.markTaskAsCompleted()
task.deleteTask()
```

Dan hoef je maar op 1 plek je DAL (data access  layer) te fixen. je aan zelfs overwegen om je Data store daarna up te daten vanuit die klasse maar dan mix je wel wat concerns.